app/src/main/java/com/android/bookswap/ui/bookAddition/BookAdditionChoice.kt

```diff
@@ -0,0 +1,139 @@
+package com.android.bookswap.ui.bookAddition
+
+import androidx.compose.foundation.BorderStroke
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.*
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.automirrored.filled.ArrowForward
+import androidx.compose.material.icons.filled.Add
+import androidx.compose.material.icons.filled.ArrowForward
+import androidx.compose.material3.Button
+import androidx.compose.material3.ButtonDefaults
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Icon
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.material3.TopAppBar
+import androidx.compose.material3.TopAppBarDefaults
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.painter.Painter
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.platform.LocalConfiguration
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.compose.ui.unit.times
+import com.android.bookswap.R
+import com.android.bookswap.ui.navigation.BackButton
+import com.android.bookswap.ui.navigation.NavigationActions
+import com.android.bookswap.ui.profile.ProfileIcon
+import com.android.bookswap.ui.theme.ColorVariable
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun BookAdditionChoiceScreen(navController: NavigationActions) {
+  val columnPadding = 16.dp
+  val buttonWidth = (LocalConfiguration.current.screenWidthDp.dp * (0.75f))
+  Scaffold(
+      topBar = {
+        TopAppBar(
+            title = { Text("Book Addition Choice", color = ColorVariable.BackGround) },
+            navigationIcon = { BackButton(navController) },
+            actions = { ProfileIcon() },
+            colors = TopAppBarDefaults.topAppBarColors(containerColor = ColorVariable.BackGround))
+      },
+      bottomBar = {
+        // BottomNavigationBar
+      }) { innerPadding ->
+        Column(
+            modifier =
+                Modifier.padding(innerPadding)
+                    .fillMaxSize()
+                    .background(ColorVariable.BackGround)
+                    .padding(bottom = 10f * columnPadding, top = columnPadding),
+            verticalArrangement = Arrangement.Center,
+            horizontalAlignment = Alignment.CenterHorizontally) {
+              ButtonWithIcon(
+                  text = "Manually",
+                  leftIcon = Icons.Default.Add,
+                  leftIconPainter = null,
+                  rightIcon = Icons.AutoMirrored.Filled.ArrowForward,

```

Comment:
here you could use a constant we because you reuse  3 times

Reply:
but those are parameters passed to a class, I can't make a variable out of a parameter, and each one has a different content, i.e, in line 64 it passes "Manually", but in line 73, passes "From ISBN". While text is the parameter which is assigned that value, this is when creating a ButtonWithIcon, a class I created further down the file.